### ISSUE TITLE
[Bug]: Duplicate images and tainted state

### ISSUE THREAD

**Issue by Blitheness** [hetznercloud/terraform-provider-hcloud#740]
### What happened?

Can't create an instance specifying "debian-12" because there are 2 images with the same name (x86/arm)

When I create an instance specifying the image ID for debian-12 arm, 114690389, every time I modify the configuration it recreates the entire instance because it is "tainted" (the provider seems to be reading back "debian-12" as the image and not equating it to the ID)

> # hcloud_server.********* must be replaced
> ...
> ~ image                      = "debian-12" -> "114690389" # forces replacement

Even though the provider supports IDs for images, It seems to be getting back "debian-12" from Hetzner and not comparing the IDs, causing the instance to be recreated.

### What did you expect to happen?

TF provider can know the image is not different between the local state and remote

### Please provide a minimal working example

```tf
resource "hcloud_server" "example_server" {
  name        = "example_server.example.org"
  server_type = "cax11"
  location    = "fsn1"
  image       = 114690389
}
```

**Comment by apricote** [comment]
Hey @Blitheness,

this issue was previously reported https://github.com/hetznercloud/terraform-provider-hcloud/issues/657 and fixed since version [v1.38.2](https://github.com/hetznercloud/terraform-provider-hcloud/releases/tag/v1.38.2). Please try to upgrade the Terraform Provider to the latest version and report back if it fixed the issue.

**Comment by Blitheness** [comment]
It is working for me in v1.42.1, thank you very much.
